Location.py

```diff
@@ -67,13 +67,17 @@ def set_rule(self, lambda_rule):
         self.access_rules = [lambda_rule]
 
 
-    def can_fill(self, state, item, check_access=True):
+    def can_fill(self, state, item, check_access=True, extra_location_checks=[]):
         if self.minor_only and item.majoritem:
             return False
-        return (
-            not self.is_disabled() and 
-            self.can_fill_fast(item) and
-            (not check_access or state.search.spot_access(self, 'either')))
+        if self.is_disabled() or not self.can_fill_fast(item) or (check_access and not state.search.spot_access(self, 'either')):
+            return False
+        if len(extra_location_checks) == 0:
+            return True
+        search_with_this = state.search.copy()
+        search_with_this.collect(item)
+        search_with_this.collect_locations(search_with_this.progression_locations() + extra_location_checks)
+        return all(search_with_this.visited(extra_location) for extra_location in extra_location_checks)
```

Comment:
```suggestion
        search_with_this.collect_locations(chain(search_with_this.progression_locations(), extra_location_checks))
        return all(map(search_with_this.visited, extra_location_checks))
```
You'll need `from itertools import chain` at the top if you take this suggestion.